tests/raftstore/transport_simulate.rs

```diff
@@ -361,3 +362,53 @@ impl FilterFactory for DropSnapshot {
         vec![box SnapshotFilter { drop: AtomicBool::new(false) }]
     }
 }
+
+/// Pause Snap
+pub struct PauseFirstSnap {
+    dropped: AtomicBool,
+    pending_msg: Mutex<Vec<StoreMsg>>,
+}
+
+impl PauseFirstSnap {
+    pub fn new() -> PauseFirstSnap {
+        PauseFirstSnap {
+            dropped: AtomicBool::new(false),
+            pending_msg: Mutex::new(vec![]),
+        }
+    }
+}
+
+impl Filter<StoreMsg> for PauseFirstSnap {
+    fn before(&self, msgs: &mut Vec<StoreMsg>) {
+        let mut to_send = vec![];
+        let mut pending_msg = self.pending_msg.lock().unwrap();
+        for m in msgs.drain(..) {
+            let paused = match m {
+                StoreMsg::ReportSnapshot { ref status, .. } => *status == SnapshotStatus::Finish,
+                StoreMsg::RaftMessage(ref msg) => {
+                    msg.get_message().get_msg_type() == MessageType::MsgSnapshot
+                }
+                _ => false,
+            };
+            if paused {
+                self.dropped.compare_and_swap(false, true, Ordering::Relaxed);
+                pending_msg.push(m);
+            } else {
+                to_send.push(m);
+            }
+        }
+        if pending_msg.len() > 1 {
```

Comment:
Each node has a different filter, hence MsgSnapshot and SnapshotStatus will appear in different filters.